### DISCUSSION TITLE
Support for both vue 2 and vue 3

### DISCUSSION THREAD

**Discussion by jourdain** [Kitware/trame#232] [Announcements]
Since `trame-client>=2.7.0` and `trame-server>=2.9.0` we introduced support for both vue 2 and 3.

__Q:__ What does it mean for you?
__A:__ That means you can start using trame with vue 3 components.

__Q:__ Can I just switch without changing any code of my current application?
__A:__ Yes if you are only using the html widgets and the vtk/paraview ones. If you are using Vuetify, you will need to use the `vuetify3` namespace which provide a slightly different API as they provide more control than before. But all the logic, syntax and approach remain exactly the same.

__Q:__ Can I use any existing trame widgets?
__A:__ Not really, so far we've only ported the client, trame, vuetify(3), vtk and paraview. Ideally we'll upgrade the main ones to be available in a cross version manner. But help from the community or suggestion on which one should go first would be appreciated.

__Q:__ How do I switch from vue2 to vue3?
__A:__ You need to set it at the very beginning on the server by calling `server.client_type = "vue3"`. By default for `trame<3` it will be "vue2" but it could be a good manner to define it in your application. 

__Q:__ Will `trame>=3` switch to "vue3" by default?
__A:__ Yes but I'm not expecting to release __trame==3.0.0__ to happen before another 6 months or so. Also trame 3 will only install `trame-client` and `trame-server` by default which means your application will be responsible to list all the widgets dependencies you aim to use.

**Comment by omdaniel** [comment]
Trame is now "vue3" by default; however, all of the examples seem to break now unless you manually revert to server.client_type = "vue2", are the examples going to be updated for "vue3"

**Reply by omdaniel** [reply]
From examples\02_vuetify\01_menu.py I tried to change the imports to 
from trame.app import get_server
from trame.widgets import vuetify3 as vuetify
from trame.ui.vuetify3 import SinglePageLayout

I was thinking any API changes would trigger an exception on the Python side; however, the webpage just fails to render correctly, and the terminal gives a JS Error

 JS Error => Lookup error: Vue.use(Vuetify.createVuetify(), {})

**Reply by omdaniel** [reply]
Perhaps, the examples should all be modified to set server.client_type = "vue2" on the official repo as a stopgap until they are updated for "vue3"

**Reply by jourdain** [reply]
Yes, those example should fix their client_type to __vue2__ until we truly migrate them. 
We fixed a bunch of them (to be vue2 explicit) but we obviously missed some. 

I will fix them shortly.

**Reply by jourdain** [reply]
Fixed in https://github.com/Kitware/trame/commit/41c78bf10a64b172b20aaa7d75ad0ca5ed40150f
